clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -57,6 +59,10 @@ public ByteBuffer serialize(ApiKeys apiKey, short version, int correlationId) {
         return Collections.singletonMap(error, 1);
     }
 
+    protected Map<Errors, Integer> errorCounts(Stream<Errors> errors) {
+        return errors.collect(Collectors.groupingBy(e -> e, Collectors.summingInt(e -> 1)));
+    }
+
     protected Map<Errors, Integer> errorCounts(Collection<Errors> errors) {
         Map<Errors, Integer> errorCounts = new HashMap<>();
         for (Errors error : errors)
```

Comment:
We can probably update `updateErrorsCounts` to use `getOrDefault` (I can't comment on the relevant line directly).